src/main/java/mSearch/filmlisten/WriteFilmlistJson.java

```diff
@@ -22,112 +22,163 @@
+                            if (datenFilm.arr[m].equals(sender)) {
+                                jg.writeString("");
+                            } else {
+                                sender = datenFilm.arr[m];
+                                jg.writeString(datenFilm.arr[m]);
+                            }
+                        } else if (m == DatenFilm.FILM_THEMA) {
+                            if (datenFilm.arr[m].equals(thema)) {
+                                jg.writeString("");
+                            } else {
+                                thema = datenFilm.arr[m];
+                                jg.writeString(datenFilm.arr[m]);
+                            }
                         } else {
-                            thema = datenFilm.arr[m];
                             jg.writeString(datenFilm.arr[m]);
                         }
-                    } else {
-                        jg.writeString(datenFilm.arr[m]);
                     }
+                    jg.writeEndArray();
                 }
-                jg.writeEndArray();
+                jg.writeEndObject();
+                Log.sysLog("   --> geschrieben!");
             }
-            jg.writeEndObject();
-            Log.sysLog("   --> geschrieben!");
         } catch (Exception ex) {
             Log.errorLog(846930145, ex, "nach: " + datei);
-        } finally {
-            try {
-                if (jg != null) {
-                    jg.close();
-                }
-                // die werden nicht immer korrekt geschlossen !??!
-                if (zipOutputStream != null) {
-                    zipOutputStream.close();
-                }
-                if (xZOutputStream != null) {
-                    xZOutputStream.close();
-                }
-            } catch (Exception e) {
-                Log.errorLog(732101201, e, "close stream: " + datei);
+        }
+    }
+
+    private Path testNativeXz() {
+        final String[] paths = {"/usr/bin/xz", "/opt/local/bin/xz", "/usr/local/bin/xz"};
+
+        Path xz = null;
+
+        for (String path : paths) {
+            //System.out.println("Path: " + path);
+            xz = Paths.get(path);
+            if (Files.isExecutable(xz)) {
+                //System.out.println("FOUND...");
+                break;
             }
         }
+
+        return xz;
     }
 
+    private void compressFile(String inputName, String outputName) throws IOException {
+        try (InputStream input = new FileInputStream(inputName);
+             FileOutputStream fos = new FileOutputStream(outputName);
+             final OutputStream output = new XZOutputStream(fos, new LZMA2Options());
+             final ReadableByteChannel inputChannel = Channels.newChannel(input);
+             final WritableByteChannel outputChannel = Channels.newChannel(output)) {
+
+            fastChannelCopy(inputChannel, outputChannel);
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)